src/fdc/D88DiskImage.cc

```diff
@@ -0,0 +1,350 @@
+			continue;
+		}
+
+		write(12, 0x00); // sync
+
+		write( 3, 0xA1);                 	// addr mark (1)
+		output.write(idx++, 0xFE, true); 	//           (2) add idam
+		output.write(idx++, sct.c); 		// C: Cylinder number
+		output.write(idx++, sct.h);  		// H: Head Address
+		output.write(idx++, sct.r); 		// R: Record
+		output.write(idx++, sct.n);  		// N: Number (length of sector: 512 = 128 << 2)
+		uint16_t addrCrc = output.calcCrc(idx - 8, 8);
+		if (sct.status == D88_STATUS_ID_CRC_ERROR)
+		{
+			addrCrc = ~addrCrc;
+		}
+		output.write(idx++, narrow_cast<uint8_t>(addrCrc >> 8));   // CRC (high byte)
+		output.write(idx++, narrow_cast<uint8_t>(addrCrc & 0xff)); //     (low  byte)
+		
+		write(22, 0x4E); // gap2
+
+		if (sct.status == D88_STATUS_NO_DATA_MARK)
+		{
+			continue;
+		}
+
+		write(12, 0x00); // sync
+		write( 3, 0xA1); // data mark (1)
+		if (sct.deleteFlag == 0)
+		{
+			write( 1, 0xFB); //           (2)
+		}
+		else
+		{
+			write( 1, 0xF8); //           (2)
+		}
+		output.writeBlock(idx, trackDataSpan.subspan(sct.dataOffset, sct.dataSize));
+		idx += sct.sectorSize;
+
+		uint16_t dataCrc = output.calcCrc(idx - (sct.sectorSize + 4), sct.sectorSize + 4);
+		if (sct.status == D88_STATUS_DATA_CRC_ERROR)
+		{
+			dataCrc = ~dataCrc;
+		}
+		output.write(idx++, narrow_cast<uint8_t>(dataCrc >> 8));   // CRC (high byte)
+		output.write(idx++, narrow_cast<uint8_t>(dataCrc & 0xff)); //     (low  byte)
+
+		write(84, 0x4E); // gap3
+	}
+	//write(182, 0x4E); // gap4b
+}
+
+void D88DiskImage::writeTrackImpl(uint8_t track, uint8_t side, const RawTrack& input)
+{
+	throw WriteProtectedException("Not Supported.");
+	// assert(side < 2);
+	// if (singleSided && (side != 0)) {
+	// 	return;
+	// }
+	// if (numTracks <= track) [[unlikely]] {
+	// 	extendImageToTrack(track);
+	// }
+	// doWriteTrack(track, side, input);
+}
+
+// SectorAccessibleDisk
+void D88DiskImage::readSectorImpl(size_t logicalSector, SectorBuffer& buf)
+{
+	auto [track, side, sector] = logToPhys(logicalSector);
+	 RawTrack rawTrack;
+	 readTrack(track, side, rawTrack);
+
+	 if (auto sectorInfo = rawTrack.decodeSector(sector)) {
+	 	// TODO should we check sector size == 512?
+	 	//      crc errors? correct track/head?
+	 	rawTrack.readBlock(sectorInfo->dataIdx, buf.raw);
+	 } else {
+	 	throw NoSuchSectorException("Sector not found");
+	 }
+}
+
+void D88DiskImage::writeSectorImpl(size_t logicalSector, const SectorBuffer& buf)
+{
+	throw WriteProtectedException("Not Supported.");
+	// auto [track, side, sector] = logToPhys(logicalSector);
+	// RawTrack rawTrack;
+	// readTrack(track, side, rawTrack);
+
+	// if (auto sectorInfo = rawTrack.decodeSector(sector)) {
+	// 	// TODO do checks? see readSectorImpl()
+	// 	rawTrack.writeBlock(sectorInfo->dataIdx, buf.raw);
+	// } else {
+	// 	throw NoSuchSectorException("Sector not found");
+	// }
+
+	// writeTrack(track, side, rawTrack);
+}
+
+size_t D88DiskImage::getNbSectorsImpl() const
+{
+	size_t t = singleSided ? 80 : (2 * 80);

```

Comment:
Very much a detail:
Maybe it's clearer to write this as: `80 * (singleSided ? 1 : 2)`